src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -93,9 +93,13 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     if (JsonUtil.equalsOrdered(compareDoc, replaceDocument())) {
       return false;
     }
-    // remove all data and add _id as first field
+    // remove all data and add _id as first field; either from original document or from replacement
     docToUpdate.removeAll();
-    if (idNode != null) docToUpdate.set(DocumentConstants.Fields.DOC_ID, idNode);
+    if (idNode != null) {
+      docToUpdate.set(DocumentConstants.Fields.DOC_ID, idNode);
+    } else if (replaceDocumentId != null) {
+      docToUpdate.set(DocumentConstants.Fields.DOC_ID, replaceDocumentId);
```

Comment:
One potential issue here: `findOneAndReplace()` shouldn't allow changing the document's id. So if `_id` is specified in the replacement document and it is exactly equal to the existing document's `_id` (if there is an existing document), then `findOneAndReplace()` should succeed. However, if `_id` is specified in the replacement document and it is different than the existing document's `_id`, `findOneAndReplace()` should fail.

Reply:
Yes: I think existing test `byIdWithDifferentId()` (added by @maheshrajamani ) catches this case, verifies it is caught and handled as expected.